syntax = "proto2";

package tutorial;

message Person {
  required string name = 1;
  required int32 id = 2;
  optional string email = 3;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }

  message PhoneNumber {
    required string number = 1;
    optional PhoneType type = 2 [default = HOME];
  }

  repeated PhoneNumber phones = 4;

  enum BuildingType {
    private = 0;
    flat = 1;
  }

  message Address {
    optional string street = 1;
    optional string number = 2;
    optional BuildingType type = 3 [default = private];
  }

  repeated Address addresses = 5;
}

message AddressBook {
  repeated Person people = 1;
}
